### PR TITLE
Build E2E tests in workflow

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #2648

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -52,7 +52,14 @@ jobs:
       - name: Build ibc-go
         run: GOARCH=${{ matrix.go-arch }} LEDGER_ENABLED=false make build
       - name: Build e2e
-        run: (cd e2e; GOARCH=${{ matrix.go-arch }} go build ./...)
+        run: |
+          cd e2e
+          test_dirs="$(ls -A tests)"
+          for td in $test_dirs
+          do
+            GOARCH=${{ matrix.go-arch }} go test -c "./tests/${td}"
```
Comment (chatton) [inline .github/workflows/test.yml:60]: `go test -c` compiles the tests without running them. The previous command was actually only building non `_test.go` code so was only catching some errors.

Comment (chatton) [reply .github/workflows/test.yml:60]: It is also only possible to run `test -c` on a single directory at a time. It's not possible to run `go test -c ./...`

**Review by colin-axner — APPROVED** [review]
yay! Thanks @chatton!

**Review by damiannolan — APPROVED** [review]
Nice nice!

**Pull request merged**